python/ray/package.py

```diff
@@ -0,0 +1,90 @@
+import hashlib
+from pathlib import Path
+from zipfile import ZipFile
+
+from enum import Enum
+from ray.experimental import internal_kv
+
+PKG_DIR='/tmp'
+
+class Protocol(Enum):
+    GCS = "gcs://"
+    PIN_GCS = "pin_gcs://"
+    # These two are not supported
+    S3 = "s3://"
+    GITHUB = "github://"
+
+def _xor_bytes(left, right):
+    if left and right:
+        return bytes(a ^ b for (a, b) in zip(left, right))
+    return left or right
+
+def _zip_module(path, zip_handler):
+    for from_file_name in path.glob("**/*"):
+        # We include pyc for performance
+        if from_file_name.match("*.py") or from_file_name.match("*.pyc"):
+            to_file_name = from_file_name.relative_to(path.parent)
+            zip_handler.write(from_file_name, to_file_name)
+
+def _hash_modules(path):
+    hash_val = None
+    BUF_SIZE = 4096 * 1024
+    for from_file_name in path.glob("**/*"):
+        md5 = hashlib.md5()
+        # We include pyc for performance
+        if from_file_name.match("*.py") or from_file_name.match("*.pyc"):
+            with open(from_file_name, mode="rb") as f:
+                data = f.read(BUF_SIZE)
+                if not data:
+                    break
+                md5.update(data)
+        hash_val = _xor_bytes(hash_val, md5.digest())
+    return hash_val
```

Comment:
This is fine for now, but we'll need to fix up the hashing later to handle larger directories, e.g., https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/_private/util.py#L260

Let's add a TODO to clean up the path hashing here (and remove xor_bytes etc).